File: ruby_squares/ruby_squares_solution.js
function getSquares(numArray) {
	// Initialize an empty array to store the perfect squares
  var squares = [];
    // Iterate over the argument array that was passed into the function
    for (var i = 0 ; i < numArray.length ; i++) {
        // Store the value of the element at index i in a variable
        var value = (numArray[i]);
        // Take the square root of the value and store it in a variable
        var squareRoot = Math.sqrt(value);
        // Turn that number (whole or decimal) into a string and split it on the decimal point
        var stringNum = squareRoot.toString().split(".");
        // Take the first element after the split, or index 0, and turn it back into an interger
        var backToInt = parseInt(stringNum[0]);
        // If the number times the number is equal to the value of the element in the array AND it is not already in the squares array
        if ((backToInt * backToInt === numArray[i]) && (squares.indexOf(numArray[i]) === -1)) {
            // push the number into the squares array
            squares.push(numArray[i])
  console.log(squares);
        }
    }
    // Return the squares array, sorted in ascending order using the comparison function that comes with .sort
    console.log(squares.sort(function(a, b) {return a - b}));
    return squares.sort(function(a, b) {
      return a - b
    });
}

getSquares([4,1,16,1,10,35,22]);

File: ruby_squares/ruby_squares_solution.rb
require "pry"

def get_squares(array_of_nums)
	# Setup an empty array to store the perfect squares
  squares_array = []

  # Iterate over each number in the array and store it in a new array
	array_of_nums.map do |num|
		# Take the square root of each number in the array
    decimals = Math.sqrt(num)
		# Take the square root of each number in the array and convert it to an integer
    whole_numbers = Math.sqrt(num).to_i

		# if the decimal is equal to the whole number (2 == 2.0) AND the number is not already in the squares array
    if decimals == whole_numbers && squares_array.index(num) == nil
			# push the number into the squares array
      squares_array.push(num)
		end

	end
    puts squares_array.sort_by { |i| i.to_i }
		# puts squares_array.sort_by(&:to_i)
end

get_squares([4,1,16,1,10,35,22])
puts "~~~"
get_squares((1..16))
puts "~~~"
get_squares((1..100))

# binding.pry

# [1, 2, 3].map(&:to_s) known as Symbol#to_proc
# is the shorthand for...
# [1, 2, 3].map { |i| i.to_s }
